data/src/history.rs

```diff
@@ -397,6 +449,90 @@ impl History {
     }
 }
 
+/// Insert the incoming message into the provided vector, sorted
+/// on server time
+///
+/// Deduplication is only checked +/- 1 second around the server time
+/// of the incoming message. Either message IDs match, or server times
+/// have an exact match + target & content.
+pub fn insert_message(messages: &mut Vec<Message>, message: Message) {
+    if messages.is_empty() {
+        messages.push(message);
+
+        return;
+    }
+
+    let start = message::fuzz_start_server_time(message.server_time);
+    let end = message::fuzz_end_server_time(message.server_time);
+
```

Comment:
I don't think these helpers are helpful or necessary. We can add / sub `chrono::Duration` from `DateTime<Utc>` and construct is `const`. So we can add `const FUZZ_SECONDS: chrono::Duration = chrono::Duration::seconds(1)` to the first line of this function and then `let start = message.server_time - FUZZ_SECONDS;`.

Reply:
Aha!  Yes, I think I added the functions primarily because I was only aware of the `TimeDelta` version of changing the time which is considerably more of a mouthful.  Secondarily because `FUZZ_SECONDS` was being shared with the `chathistory` request.  I don't remember why I did that, and while they're not completely unrelated I don't see any reason to keep them in lock-step.  I'll add `FUZZ_SECONDS` here as you suggest, and set a distinct `FUZZ_SECONDS` inside `isupport.rs`.